### PR TITLE
prevent focus loss

### PR DESCRIPTION
**Pull request by jackkav**
during development, when modifying a class component hmr triggers a full-reload event on the ws connection. This event is picked up by the browserwindow will-navigate and causes chrome to steal focus. This change corrects that behaviour.

### CODE REVIEW

**Comment by dimitropoulos** [comment]
can you please share some reproduction steps?  Is `ws connection` workspace?

**Comment by jackkav** [comment]
> can you please share some reproduction steps?  Is `ws connection` workspace?

Sure if you modify ui/index.tsx you will see a full-reload event in the websocket dev tool view. In normal cases the reload will happen as usual and get a rerender but if you leave your dev env running for a while the hmr stops working and full-reload causes the will-navigate event to receive a url with an anchor # on the end which causes the handler to misinterpret it as an outside url click. This change makes the handler ignore local dev vite full-reload events.

**Review by marckong — APPROVED** [review]

**Review by dimitropoulos — DISMISSED** [review]
@jackkav in that case can you please document this in the code?  this is a pretty tough-to-discern edge-case just from looking at the code and we'd benefit a lot from including it (down the road, when we start wondering why it was using `startsWith`).  you could pretty much take what you already wrote verbatim

File: packages/insomnia/src/main/window-utils.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ export function createWindow() {
 
   // Open generic links (<a .../>) in default browser
   mainWindow?.webContents.on('will-navigate', (event, url) => {
-    if (url === appUrl) {
+    if (url.startsWith(appUrl)) {
```
Comment (filfreire) [inline packages/insomnia/src/main/window-utils.ts:108]:
```suggestion
    // Prevents local dev full-reload events from opening browser window, see https://github.com/Kong/insomnia/pull/4925
    if (url.startsWith(appUrl)) {
```

**Review by filfreire — APPROVED** [review]
Tested locally and seems to be working as intended.

Before - every change to the `ui/index.tsx` file and other files that would cause Insomnia to reload would open a `localhost` related browser tab:
Screenshot 2022-07-04 at 10 59 16

After - local changes no longer open browser tabs:

Screenshot 2022-07-04 at 11 16 06

**Comment by jackkav** [comment]
I'm not sure exactly what puts the hash on the end of the url after a certain amount of time, but I have a hacky repro.
1. set the window location `window.location='http://localhost:3334/index.html#'` in chrome devtools. 
2. Make a change to a react function component. 

Before HMR would open a browser tab.
After HMR would reload only the changed function component.

**Pull request merged**